src/data/utils.py

```diff
@@ -0,0 +1,347 @@
+                rqe.append(frame["I3Calibration"].dom_cal[om_key].relative_dom_eff)
+                x.append(gcd_dict[om_key].position.x)
+                y.append(gcd_dict[om_key].position.y)
+                z.append(gcd_dict[om_key].position.z)
+        
+    features = {'charge': charge, 
+                'dom_time': time, 
+                'dom_x': x, 
+                'dom_y': y, 
+                'dom_z': z,
+                'width' : width,
+                'pmt_area': area, 
+                'rde': rqe}
+    return features
+
+def build_retroreco_extraction(is_mc):
+    ''' Builds a default RetroReco extraction dictionary. Contains associated quantities, such as BDT classifiers.
+
+        is_mc               : boolean, if True the estimated weight of the simulated event is extracted too.
+
+        RETURNS
+        retro_extraction    : dictionary where each field contains a string that is evaluated later using eval()
+
+    '''
+    retro_extraction = {'azimuth_retro': 'frame["L7_reconstructed_azimuth"].value',
+                        'time_retro': 'frame["L7_reconstructed_time"].value',
+                        'energy_retro': 'frame["L7_reconstructed_total_energy"].value', 
+                        'position_x_retro': 'frame["L7_reconstructed_vertex_x"].value', 
+                        'position_y_retro': 'frame["L7_reconstructed_vertex_y"].value',
+                        'position_z_retro': 'frame["L7_reconstructed_vertex_z"].value',
+                        'zenith_retro': 'frame["L7_reconstructed_zenith"].value',
+                        'azimuth_sigma': 'frame["L7_retro_crs_prefit__azimuth_sigma_tot"].value',
+                        'position_x_sigma': 'frame["L7_retro_crs_prefit__x_sigma_tot"].value',
+                        'position_y_sigma': 'frame["L7_retro_crs_prefit__y_sigma_tot"].value',
+                        'position_z_sigma': 'frame["L7_retro_crs_prefit__z_sigma_tot"].value',
+                        'time_sigma': 'frame["L7_retro_crs_prefit__time_sigma_tot"].value',
+                        'zenith_sigma': 'frame["L7_retro_crs_prefit__zenith_sigma_tot"].value',
+                        'energy_sigma': 'frame["L7_retro_crs_prefit__energy_sigma_tot"].value',
+                        'cascade_energy_retro': 'frame["L7_reconstructed_cascade_energy"].value',
+                        'track_energy_retro': 'frame["L7_reconstructed_track_energy"].value',
+                        'track_length_retro': 'frame["L7_reconstructed_track_length"].value',
+                        'L7_MuonClassifier_FullSky_ProbNu': 'frame["L7_MuonClassifier_FullSky_ProbNu"].value',
+                        'L4_MuonClassifier_Data_ProbNu': 'frame["L4_MuonClassifier_Data_ProbNu"].value',
+                        'L4_NoiseClassifier_ProbNu': 'frame["L4_NoiseClassifier_ProbNu"].value',
+                        'L7_PIDClassifier_FullSky_ProbTrack': 'frame["L7_PIDClassifier_FullSky_ProbTrack"].value'}
```

Comment:
Good to know! 

I did indeed extract these quantities like extract_features in the past, but my experience is that it is a quite annyoing format for when you want to change the extracted quantities. Also, the dictionary field names ends up being the column names in the sqlite database. So for swiftness, a user could pass something like the above and it would work flawlessly.

Right now, when an evaluation fails, it simply pads the failed value with default -1.